### ISSUE TITLE
ppx_expect + reason is broken after dune lang 3.2

### ISSUE THREAD

**Issue by hackwaly** [ocaml/dune#7930]
## Expected Behavior
```
File "test/Test.re", line 1, characters 0-0:
diff --git a/_build/default/test/Test.re b/_build/default/test/Test.re.corrected
index e3f7c74..d69e228 100644
--- a/_build/default/test/Test.re
+++ b/_build/default/test/Test.re.corrected
@@ -1,4 +1,4 @@
 let%expect_test "test" = {
   print_endline("hello");
-  [%expect{| |}]
+  [%expect{| hello |}]
 }
```

## Actual Behavior
```
File "test/Test.re", line 1:       
Error: I/O error: test/Test.re: No such file or directory
```

## Reproduction

test/dune
```dune
(library
 (name test)
 (inline_tests)
 (preprocess (pps ppx_expect))
 (libraries ))
```

test/test.re
```reason
let%expect_test "test" = {
  print_endline("hello");
  [%expect{| |}]
}
```

dune-project
```
(lang dune 3.3)
```

If change the line `(lang dune 3.3)` to `(lang dune 3.2)`. Everything seems ok.

## Specifications

- Version of `dune` (output of `dune --version`): 3.8.1
- Version of `ocaml` (output of `ocamlc --version`): 4.14.1
- Operating system (distribution and version): macOS 13.4 (22F66)

**Comment by Alizter** [comment]
This appears to be a bug with how we are sandboxing ppx. It appears the dialect of OCaml is ignored. See #7932 for a reproduction case.

**Comment by Alizter** [comment]
Fix is also in #7932
